DesignSafe-CI/portal
# task/DES-1694: Published Project Feedback Form

## Overview: ##
Adds a "Leave Feedback" button link to published project views. Submitted feedback generates an RT ticket to the `Designsafe-pub-feedback` queue.

## PR Status: ##


## Related Jira tickets: ##

* [DES-1694](https://jira.tacc.utexas.edu/browse/DES-1694)

## Summary of Changes: ##

- "Leave Feedback" button added, which opens a feedback modal
- Expanded the `Django` angular constant to add more user data
- Expanded the `rtUtil` class to allow a `queue` to be passed

## Testing Steps: ##
1. Open a Publication in dev, like https://designsafe.dev/data/browser/public/designsafe.storage.published/PRJ-2805
2. Click "Leave Feedback"
3. Enter data and submit
4. Confirm feedback ends up in the [RT queue](https://consult.tacc.utexas.edu/Search/Results.html?Query=Queue%20%3D%20%27Designsafe-pub-feedback%27%20AND%20Status%20%3D%20%27__Active__%27)
5. Ensure the button exists for every published project type 

## UI Photos:

<img width="294" alt="Screen Shot 2021-06-04 at 12 20 54 AM" src="https://user-images.githubusercontent.com/20326896/120750309-c2c49380-c4cb-11eb-9413-1f4131aa3c68.png">
<img width="613" alt="Screen Shot 2021-06-04 at 12 21 23 AM" src="https://user-images.githubusercontent.com/20326896/120750303-c22bfd00-c4cb-11eb-9bed-e3568cb8eb7a.png">
<img width="625" alt="Screen Shot 2021-06-04 at 12 21 06 AM" src="https://user-images.githubusercontent.com/20326896/120750304-c22bfd00-c4cb-11eb-8cbd-b4d8c837ad0f.png">
